crates/ty_python_semantic/src/program.rs

```diff
@@ -118,6 +162,54 @@ pub enum PythonVersionSource {
     Default,
 }
 
+impl PythonVersionSource {
+    fn priority(&self) -> PythonSourcePriority {
+        match self {
+            PythonVersionSource::Default => PythonSourcePriority::Default,
+            PythonVersionSource::PyvenvCfgFile(_) => PythonSourcePriority::PyvenvCfgFile,
+            PythonVersionSource::ConfigFile(_) => PythonSourcePriority::ConfigFile,
+            PythonVersionSource::Cli => PythonSourcePriority::Cli,
+        }
+    }
+}
+
+/// The priority in which Python version sources are considered.
+/// A higher value means a higher priority.
+///
+/// For example, if a Python version is specified in a pyproject.toml file
+/// but *also* via a CLI argument, the CLI argument will take precedence.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord)]
+enum PythonSourcePriority {
+    Default = 0,
+    PyvenvCfgFile = 1,
+    ConfigFile = 2,
+    Cli = 3,
+}
```

Comment:
```suggestion
enum PythonSourcePriority {
    Default,
    PyvenvCfgFile,
    ConfigFile,
    Cli,
}
```

Reply:
I know they're not necessary, but I kinda prefer having them there so that it's explicit that `Cli` is definitely meant to have a higher priority than `ConfigFile` (for example)... I guess I should add a test for that as well!